content/blog/simplify-embedded-projects-with-esp-bsp/index.md

```diff
@@ -0,0 +1,134 @@
+---
+title: "Simplify Your Embedded Projects with ESP-BSP"
+date: 2024-06-18
+showAuthor: false
+authors:
+    - "juraj-michalek"
+tags: ["Embedded Systems", "ESP32", "ESP32-S3", "Espressif", "BSP"]
+---
+
+# Simplify Your Embedded Projects with ESP-BSP
+
+## Introduction
+
+Are you a maker or an embedded systems enthusiast looking to create applications that work seamlessly across different ESP32 development boards? Whether you’re using the [ESP-WROVER-KIT](https://docs.espressif.com/projects/esp-idf/en/stable/esp32/hw-reference/esp32/get-started-wrover-kit.html), [M5Stack-CoreS3](https://docs.m5stack.com/en/core/CoreS3), [ESP32-S3-BOX-3](https://github.com/espressif/esp-box/blob/master/docs/hardware_overview/esp32_s3_box_3/hardware_overview_for_box_3.md), or other compatible boards, the ESP Board Support Package (ESP-BSP) makes your life easier. In this article, we’ll walk you through how to get started with ESP-BSP, enabling you to focus on your project’s functionality without worrying about hardware differences.
+
+## What is ESP-BSP?
+
+[ESP-BSP](https://github.com/espressif/esp-bsp) is a collection of Board Support Packages featuring Espressif’s or M5Stack development boards. The set can be extended to any board with Espressif’s chip. It provides a convenient way to start a project for a specific development board without the need to manually look for drivers and other details. By using ESP-BSP, you can:
+
+- **Streamline Hardware Integration**: Simplify code and reduce complexity.
+- **Enhance Development Speed**: Quickly set up and start working on your projects.
+- **Access Standardized APIs**: Ensure consistency across your projects.
+
+## Getting Started with ESP-BSP
+
+The following example covers steps for building application for ESP32-S3-BOX-3 which is supported by ESP-BSP.
+
+### Hardware Setup
+
+Ensure you have the following hardware:
+
+- [ESP32-S3-BOX-3](https://github.com/espressif/esp-box) development board.
+- USB-C Cable for power and programming.
+
+### Prerequisites
+
+Before you begin, make sure you have the following:
+
+- [ESP-IDF v5.3](https://docs.espressif.com/projects/esp-idf/en/stable/esp32/get-started/index.html#installation): The official development framework for the ESP32, properly installed and sourced in your shell.
```

Comment:
IDF v5.3 has not been released yet, and the `stable` link doesn't necessarily link to IDF v5.3. It points to v5.2.2 now, it will point to v5.3 at some point, and it might point to v5.4 in the future.

I suggest updating the link to point a fixed release and updating the link text to match that release.

Reply:
Good point. Updated to release v5.3. I expect to update the article in the future, e.g. to remove the patch parts with esp_codec_dev, once BSP gets fixed.